lib/monaco-adapter.js

```diff
@@ -285,7 +285,7 @@ var MonacoAdapter = function () {
    */
   MonacoAdapter.prototype.registerRedo = function registerRedo(callback) {
     if (typeof callback === 'function') {
-      this.callbacks.redo = callback;
+      this.monacoModel.redo = callback;
```

Comment:
looks good!

Btw, how does it work exactly, do we need to instantiate a `MonacoAdapter` per monaco model?

Reply:
I don't think so, because we are currently disposing of `firepad` and instantiating a new one whenever the monaco model changes.